text/0000-code-literals.md

```diff
@@ -0,0 +1,268 @@
+functionality as required.
+
+If it is necessary to include triple backticks within a code string
+literal, more than three backticks may be used to enclose the
+literal, eg.
+
+```rust
+let code = ````
+    ```
+````;
+```
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+A code string literal will begin and end with three or more backticks.
+The number of backticks in the terminator must match the number used
+to begin the literal.
+
+The value of the string literal will be determined using the following
+steps:
+
+1.  Start from the first newline after the opening backticks.
+2.  Take the string exactly as written until the closing backticks.
+3.  Remove equal numbers of spaces or tabs from every non-empty line
+    until the first character of the first non-empty line is neither
+    a space nor a tab, or until every line is empty.
+    Raise a compile error if this could not be done
+    due to a "negative" indent or inconsistent whitespace (eg. if
+    some lines are indented using tabs and some using spaces).
+
+Here are some edge case examples:
+
+```rust
+    // Empty string
+    assert_eq!(```foo
+    ```, "");
+
+    // Newline
+    assert_eq!(```
+
+    ```, "\n");
+
+    // No terminating newline
+    assert_eq!(```
+        bar```, "bar");
+
+    // Terminating newline
+    assert_eq!(```
+        bar
+    ```, "bar\n");
+
+    // Preserved indent
+    assert_eq!(```
+    if a:
+        print(42)
+    ```, "if a:\n    print(42)\n");
+
+    // Relative indent
+    assert_eq!(```
+            if a:
+                print(42)
+    ```, "if a:\n    print(42)\n");
+
+    // Relative to first non-empty line
+    assert_eq!(```
+
+
+            if a:
+                print(42)
+    ```, "\n\nif a:\n    print(42)\n");
+```
+
+The text between the opening backticks and the first newline is
+preserved within the AST, but is otherwise unused.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+The main drawback is increased complexity of the language:
+
+1. It adds a new symbol to the language, which was not previously used.
+2. It adds a third way of writing string literals.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+There is lots of room to bike-shed syntax.
+If there is significant opposition to the backtick syntax, then an
+alternative syntax such as:
+```
+code"
+    string
+"
+```
+could be used.
+
+Similarly, the use of more than three backticks may be unpopular.
+It's not clear how important it is to be able to nest backticks
+within backticks, but a syntax mirroring raw string literals could
+be used instead, eg.
+```
+`# foo
+    string
+#`
+```
+
+There is also the question of whether the backtick syntax would
+interfere with the ability to paste Rust code snippets into such
+blocks. Experimentally, markdown parsers do not seem to have any
+problems with this (as demonstrated in this document).
+
+# Prior art
```

Comment:
<https://docs.rs/indoc/latest/indoc/> is probably worthy of a mention.

This would also be a good place to include mentions of language features that accomplish similar goals in other languages, like [here-doc](https://perldoc.perl.org/perlop#EOF)s.